src/actions/setAndLockDomainMetadata.ts

```diff
@@ -0,0 +1,22 @@
+import { BigNumberish, ethers } from "ethers";
+import { Registrar } from "../contracts/types";
+import { validateOwner, validateStatus } from "./helpers";
+
+export const setAndLockDomainMetadata = async (
+  domainId: BigNumberish,
+  metadataUri: string,
+  potentialOwner: string,
+  registrar: Registrar
+): Promise<ethers.ContractTransaction> => {
+  validateOwner(
+    domainId,
+    potentialOwner,
+    registrar,
+    "Cannot set metadata of unowned domain"
+  );
+  validateStatus(domainId, registrar, "Locked metadata cannot be modified");
```

Comment:
Nit: Using a bit more active language for error messages might be more clear to users; "Metadata must be unlocked to be modified"